### PR TITLE
Schouten Tensor

### LINKED ISSUE #314: Schouten Tensor
🎯 **Goal**
 
 * Create Schouten Tensor module
 * https://en.wikipedia.org/wiki/Schouten_tensor

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

### PR DESCRIPTION
**Pull request by OVSofia**
Fixes #314

### CODE REVIEW

**Comment by ritzvik** [comment]
Let's complete #322 and #317 before coming to this!

File: src/einsteinpy/symbolic/schouten.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
+from einsteinpy.symbolic.ricci import RicciScalar, RicciTensor
+from einsteinpy.symbolic.riemann import RiemannCurvatureTensor
+from einsteinpy.symbolic.tensor import BaseRelativityTensor, _change_config
+
+
+class SchoutenTensor(BaseRelativityTensor):
+    """
+    Inherits from ~einsteinpy.symbolic.tensor.BaseRelativityTensor .
+    Class for defining Schouten Tensor
+    """
```
Comment (ritzvik) [inline src/einsteinpy/symbolic/schouten.py]: @OVSofia , try leaving a blank line after ending of each docstring.
```
    """
    Inherits from ~einsteinpy.symbolic.tensor.BaseRelativityTensor .
    Class for defining Schouten Tensor

    """
```

File: src/einsteinpy/symbolic/schouten.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+        Raises
+        ------
+        ValueError
+            Raised when the dimension of the tensor is less than 3
+
+        """
+        if metric.dims >= 3:
+            t_riemann = RiemannCurvatureTensor.from_metric(metric)
+            t_ricci = RicciTensor.from_riemann(t_riemann, parent_metric=None)
```
Comment (ritzvik) [inline src/einsteinpy/symbolic/schouten.py]: What's the need for Riemann Tensor. Just call Ricci Tensor by `from_metric` classmethod.

File: src/einsteinpy/symbolic/schouten.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            r_scalar = RicciScalar.from_riccitensor(t_ricci, parent_metric=None)
+            dims = metric.dims
+            t_schouten = (
+                1
+                / (dims - 2)
+                * (
+                    t_ricci.tensor()
+                    - 1 / (2 * (dims - 1)) * r_scalar._expr * metric.tensor()
```
Comment (ritzvik) [inline src/einsteinpy/symbolic/schouten.py]: Can you replace `t_schouten` with this?
`(t_ricci.tensor() - (r_scalar.expr*metric.tensor()/(2*(dims-1))))/(dims-2)`

Comment (ritzvik) [reply src/einsteinpy/symbolic/schouten.py]: Sorry for me being too fussy 😞

File: src/einsteinpy/tests/test_symbolic/test_schouten.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        St2 = SchoutenTensor(St.tensor(), St.syms, config="uuu")
+        assert False
+    except ValueError:
+        assert True
+
+
+def test_SchoutenTensor():
+    metric = anti_de_sitter_metric()
+    st = SchoutenTensor.from_metric(metric)
+    assert st != "Tensor"
```
Comment (ritzvik) [inline src/einsteinpy/tests/test_symbolic/test_schouten.py]: after, the changes above, we can see what can we do about the tests!

File: src/einsteinpy/symbolic/schouten.py (hunk 3 above)
Comment (OVSofia) [reply src/einsteinpy/symbolic/schouten.py]: jaja Don't worry! :)

**Comment by ritzvik** [comment]
@OVSofia can you try to increase the coverage?
Give it a day, I'm feeling very lazy. I'll see to it tomorrow.

Also, rebase!

**Comment by OVSofia** [comment]
> coverage

yes, I will try tomorrow

File: src/einsteinpy/tests/test_symbolic/test_schouten.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    try:
+        St2 = SchoutenTensor(St.tensor(), St.syms, config="uuu")
+        assert False
+    except ValueError:
+        assert True
+
+
+def test_SchoutenTensor():
+    metric = anti_de_sitter_metric()
+    st = SchoutenTensor.from_metric(metric)
```
Comment (ritzvik) [inline src/einsteinpy/tests/test_symbolic/test_schouten.py]: What's this test for??. 
I don't find it useful. Maybe I am wrong.
Can you explain it??

File: src/einsteinpy/tests/test_symbolic/test_schouten.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    try:
+        obj = SchoutenTensor.from_metric(metric)
+        assert False
+    except ValueError:
+        assert True
+
+
+def test_SchoutenTensor_with_arbritary_config():
+    sch_inv = schwarzschild_metric()
+    ch = SchoutenTensor.from_metric(sch_inv)
```
Comment (ritzvik) [inline src/einsteinpy/tests/test_symbolic/test_schouten.py]: rename `sch_inv` to `sch`, I don't think it's an inverse!

Comment (ritzvik) [reply src/einsteinpy/tests/test_symbolic/test_schouten.py]: After doing the required changes, kindly squash the commits.
BTW you are doing  a great job!! 😍

File: src/einsteinpy/tests/test_symbolic/test_schouten.py (hunk 5 above)
Comment (OVSofia) [reply src/einsteinpy/tests/test_symbolic/test_schouten.py]: I did it to see that it called the function well, but now it is no longer useful. I'm going to delete it

File: src/einsteinpy/tests/test_symbolic/test_schouten.py (hunk 6 above)
Comment (OVSofia) [reply src/einsteinpy/tests/test_symbolic/test_schouten.py]: Okey, and thanks!! :D

**Comment by ritzvik** [comment]
@shreyasbapat LGTM

**Pull request merged**